x/pairing/keeper/filters/filter.go

```diff
@@ -0,0 +1,68 @@
+package filters
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	projectstypes "github.com/lavanet/lava/x/projects/types"
+)
+
+type Filter interface {
+	Filter(ctx sdk.Context, providers []epochstoragetypes.StakeEntry) []bool
+	InitFilter(strictestPolicy projectstypes.Policy) bool // return if filter is usable (by the policy)
+}
+
+func GetAllFilters() []Filter {
+	var selectedProvidersFilter SelectedProvidersFilter
+	var frozenProvidersFilter FrozenProvidersFilter
+	var geolocationFilter GeolocationFilter
+
+	filters := []Filter{&selectedProvidersFilter, &frozenProvidersFilter, &geolocationFilter}
+	return filters
+}
+
+func initFilters(filters []Filter, strictestPolicy projectstypes.Policy) []Filter {
```

Comment:
make this public

Reply:
I don't think there's a need to do this since FilterProviders calls it and that's it. You don't need to call it without FilterProviders